internal/topology/upgrade/clusterctl_upgrade_test.go

```diff
@@ -704,6 +752,92 @@ func checkOmittableField(mustBeSet bool) func(obj *unstructured.Unstructured) er
 	}
 }
 
+func checkPtrTToT() func(obj *unstructured.Unstructured) error {
+	return func(obj *unstructured.Unstructured) error {
+		switch obj.GetKind() {
+		case "TestResource":
+			value, exists, err := unstructured.NestedString(obj.Object, "spec", "ptrStringToString")
+			if err != nil {
+				return err
+			}
+			if exists && value == "" {
+				return errors.New("expected to not contain an empty ptrStringToString field")
+			}
+		case "TestResourceTemplate":
+			value, exists, err := unstructured.NestedString(obj.Object, "spec", "template", "spec", "ptrStringToString")
+			if err != nil {
+				return err
+			}
+			if exists && value == "" {
+				return errors.New("expected to not contain an empty ptrStringToString field")
+			}
+		}
+		return nil
+	}
+}
+
+func checkTToPtrT() func(obj *unstructured.Unstructured) error {
+	return func(obj *unstructured.Unstructured) error {
+		switch obj.GetKind() {
+		case "TestResource":
+			value1, exists, err := unstructured.NestedInt64(obj.Object, "spec", "int32ToPtrInt32")
+			if err != nil {
+				return err
+			}
+			if exists && value1 == 0 {
+				return errors.New("expected to not contain a zero int32ToPtrInt32 field")
+			}
+			value2, exists, err := unstructured.NestedBool(obj.Object, "spec", "boolToPtrBool")
+			if err != nil {
+				return err
+			}
+			if exists && !value2 {
+				return errors.New("expected to not contain a false boolToPtrBool field")
+			}
+		case "TestResourceTemplate":
+			value1, exists, err := unstructured.NestedInt64(obj.Object, "spec", "template", "spec", "int32ToPtrInt32")
+			if err != nil {
+				return err
+			}
+			if exists && value1 == 0 {
+				return errors.New("expected to not contain an empty int32ToPtrInt32 field")
```

Comment:
```suggestion
				return errors.New("expected to not contain a zero int32ToPtrInt32 field")
```
for consistency with above